### ISSUE TITLE
Display bug regarding hydrostorage CO2

### ISSUE THREAD

**Issue by ThomasAuriel** [electricitymaps/electricitymaps-contrib#3668]
In co2eq_parameters.json, hydro discharge and batterry discharge are specified. The value computed is the national/world average CO2 intensity.

https://github.com/electricitymap/electricitymap-contrib/blob/master/config/co2eq_parameters.json

By doing so, are not you counting two times the CO2 produced for stored Wh ? One time at the generation and one time at the discharge ?

Thank you,
Thomas

**Comment by corradio** [comment]
Hello @ThomasAuriel ,

Good to see you here. Emissions are counted during discharge only, as can be seen here:
https://github.com/electricitymap/electricitymap-contrib/blob/master/config/co2eq_parameters.json#L4-L8
No double counting!

**Comment by jarek** [comment]
I think/suspect the initial setup of storage/battery operations has been based on cross-border exchanges. Electricity Map does subtract the emissions "exported" from a region's total, so that works. But I'm not sure if the emissions "stored" are subtracted - I can't seem to see this happening in the UI.

Taking for example Austria right now, with data available for 7pm EST / 1am Austrian time: the consumption chart shows exports and pumping hydro as negative. But while the emissions chart shows exports as negative, it doesn't show pumping hydro being negative as well. So are the emissions associated with the production of energy subsequently stored in the pumped hydro being subtracted, and just not shown?

![austria 3am consumption](https://user-images.githubusercontent.com/47415/152462839-86613213-637c-4029-81ca-826e97f76bb1.png)

![austria 3am emissions](https://user-images.githubusercontent.com/47415/152462848-6d156278-4d68-44ad-b056-476b920fc5f5.png)

**Comment by Macquaria** [comment]
Thanks @jarek for illustrating the point.

The "?" figue instead of "0" for charging hydro/battery storage carbon emissions is intriguing, but I'm almost certain there's no substraction like for electricity exports. 

Actually, substracting average emissions when pumping/charging wouldn't be a good way to fix the double counting. Indeed, storage facilities have an efficiency <1 (~0.8 for PHS, ~0.9 for li-ion batteries). 
Consequence : if a pumped storage hydro plant "removes" 1 kWh x _Average carbon intensity_ when pumping, it will only release 0.8 kWh x _Average carbon intensity_, so that its overall carbon footprint will be counted as ... negative.

To me, the simple and proper way to fix the suspected double-counting would be to maintain a null toll on emissions when pumping/charging, and assign the storage system's life cycle emissions (overall indirect emissions / amount of electricity **released** over lifetime) when discharging.

**Comment by corradio** [comment]
> So are the emissions associated with the production of energy subsequently stored in the pumped hydro being subtracted, and just not shown as such, so it's a display bug? If they're not being subtracted in data, it does look like it's double-counting.

Good point, that's a display bug! They should be shown as negative just as the exchanges.

**Comment by madsnedergaard** [comment]
I'm looking into this again, but need some help on understanding what the correct visual representation should be before making the changes :)

## When storing/charging...
<img width="586" alt="Screenshot 2023-02-26 at 21 20 08" src="https://user-images.githubusercontent.com/3296643/221435070-5eb83ede-59ef-4f25-be3e-0a5ccd896d08.png">

What numbers should be displayed in the tooltip - should the question marks just be replaced with 0?

<img width="565" alt="Screenshot 2023-02-26 at 21 19 37" src="https://user-images.githubusercontent.com/3296643/221435078-e1923610-2ad1-4a33-9d81-1953442126f6.png">

The bar should be negative (on the left side of the axis) like with exports, is what I gathered from previous comments - is that correct?

## When discharging...
<img width="593" alt="Screenshot 2023-02-26 at 21 20 00" src="https://user-images.githubusercontent.com/3296643/221435102-34d202a3-ddc3-4ff3-9f59-326986e27496.png">
<img width="566" alt="Screenshot 2023-02-26 at 21 19 47" src="https://user-images.githubusercontent.com/3296643/221435108-4ea60c19-1fa4-4b1c-978a-4f50cfe9f378.png">

Can someone verify that all numbers are indeed correct here? :)
